src/main/java/seedu/bookmark/logic/commands/AddNoteCommand.java

```diff
@@ -0,0 +1,112 @@
+package seedu.bookmark.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.bookmark.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.bookmark.logic.parser.CliSyntax.PREFIX_NOTE_TEXT;
+import static seedu.bookmark.model.Model.PREDICATE_SHOW_ALL_BOOKS;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import seedu.bookmark.commons.core.Messages;
+import seedu.bookmark.commons.core.index.Index;
+import seedu.bookmark.logic.commands.exceptions.CommandException;
+import seedu.bookmark.model.Model;
+import seedu.bookmark.model.book.Book;
+import seedu.bookmark.model.book.Bookmark;
+import seedu.bookmark.model.book.Genre;
+import seedu.bookmark.model.book.Name;
+import seedu.bookmark.model.book.Note;
+import seedu.bookmark.model.book.TotalPages;
+import seedu.bookmark.model.tag.Tag;
+
+/**
+ * Adds a Book into a Library.
+ */
+public class AddNoteCommand extends Command {
+
+    public static final String COMMAND_WORD = "note";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a note to the book. \n"
+                                                       + "Parameters: INDEX (must be a positive integer) "
+                                                       + PREFIX_NAME + "TITLE "
+                                                       + PREFIX_NOTE_TEXT + "TEXT";
+
+    public static final String MESSAGE_ADD_NOTE_SUCCESS = "Added Note to Book: %1$s";
+    public static final String MESSAGE_DUPLICATE_BOOK = "This book already exists in the library.";
+    public static final String MESSAGE_DUPLICATE_NOTE = "This note already exists for the book.";
```

Comment:
Am not too sure if it is necessary to check for duplicate notes, but I think it's fine to include